packages/ember-metal/lib/meta.js

```diff
@@ -326,7 +327,7 @@ export class Meta {
     let descriptor = lookupDescriptor(obj, key);
     let isMandatorySetter = descriptor && descriptor.set && descriptor.set.isMandatorySetter;
 
-    if (isMandatorySetter) {
+    if (isMandatorySetter === true) {
```

Comment:
It doesn't, that's why you need to change the definition of `isMandatorySetter` above. :-)